### PR TITLE
feat: Perform focus and scroll when invalidated based on autoFocusOnValidationFailure and shouldScrollToInvalidField

### LINKED ISSUE #955: Disable RequestFocus when calling invalidateField
In my code I'm calling invalidateField on each field because the validation logic happens in the backend. I would like to hide the keyboard when doing this, but the invalidateField method always requests focus when called:

https://github.com/danvick/flutter_form_builder/blob/a582442f1e6bffc2fc8c4503585d03ea551c7973/packages/flutter_form_builder/lib/src/form_builder_field.dart#L228:L237

```
  void requestFocus() {
    FocusScope.of(context).requestFocus(effectiveFocusNode);
    Scrollable.ensureVisible(context);
  }

  void invalidate(String errorText) {
    setState(() => _customErrorText = errorText);
    validate(clearCustomError: false);
    requestFocus();
  }
  ```

Is it possible to disable or circumvent this behaviour?

### PR DESCRIPTION
**Pull request by pongloongyeat**
Closes #955.

Adds an additional field `shouldScrollToInvalidField` which defaults to `false`. On master, this is essentially this PR with `shouldScrollToInvalidField` set to `true`. However, due to issues such as #955, I've set it to default to `true`.

### CODE REVIEW

File: packages/flutter_form_builder/lib/src/form_builder.dart [hunk 1]
```diff
@@ -63,6 +63,9 @@ class FormBuilder extends StatefulWidget {
   /// Whether the form should auto focus on the first field that fails validation.
   final bool autoFocusOnValidationFailure;
 
+  /// Whether the form should scroll to the first field that fails validation.
+  final bool shouldScrollToInvalidField;
```
Comment (deandreamatias) [inline packages/flutter_form_builder/lib/src/form_builder.dart:67]: Maybe can rename this property to `autoScrollOnValidationFailure` or `autoScrollToInvalidField`. What do you think?

Comment (pongloongyeat) [reply packages/flutter_form_builder/lib/src/form_builder.dart:67]: I prefer `autoScrollToInvalidField` actually. Will change it out.

**Comment by pongloongyeat** [comment]
> Can be add tests please?

Sure. Will do it over the weekend.

**Comment by deandreamatias** [comment]
Nice!
Thanks a lot @pongloongyeat for your contribution 🚀

**Comment by pongloongyeat** [comment]
Getting a little lost on writing tests to detect for scrolling. Could anyone point me in the right direction?

**Comment by deandreamatias** [comment]
> Getting a little lost on writing tests to detect for scrolling. Could anyone point me in the right direction?

Hi @pongloongyeat,
Has some methods to tests scroll with widgets. Take a look to [this](https://api.flutter.dev/flutter/flutter_driver/FlutterDriver/scrollUntilVisible.html) and related

**Pull request closed without merging**